src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/TagHelperAttribute.cs

```diff
@@ -77,8 +79,130 @@ public bool Equals(TagHelperAttribute other)
             return
                 other != null &&
                 string.Equals(Name, other.Name, StringComparison.OrdinalIgnoreCase) &&
-                Minimized == other.Minimized &&
-                (Minimized || Equals(Value, other.Value));
+                Structure == other.Structure &&
+                (Structure == HtmlAttributeStructure.Minimized || Equals(Value, other.Value));
+        }
+
+        /// <inheritdoc />
+        public void WriteTo(TextWriter writer, HtmlEncoder encoder) => WriteTo(Name, Value, Structure, writer, encoder);
+
+        /// <inheritdoc />
+        public void CopyTo(IHtmlContentBuilder destination) => CopyTo(Name, Value, Structure, destination);
+
+        /// <inheritdoc />
+        public void MoveTo(IHtmlContentBuilder destination) => CopyTo(destination);
+
+        /// <summary>
+        /// TODO
+        /// </summary>
+        /// <param name="name"></param>
+        /// <param name="value"></param>
+        /// <param name="structure"></param>
+        /// <param name="destination"></param>
+        public static void CopyTo(
+            string name,
+            object value,
+            HtmlAttributeStructure structure,
+            IHtmlContentBuilder destination)
```

Comment:
`TagHelperAttribute`s are already a decently large portion of a `TagHelper` pages allocations. I'd like to avoid re-allocating new `TagHelperAttribute`s per your comment [here](https://github.com/aspnet/Mvc/pull/4587#discussion_r63645382).